### ISSUE TITLE
Plugin retrieves text but text doesn't change in Figma

### ISSUE THREAD

**Issue by DorothyLindman** [imnotashrimp/copyThat-airtable-figma#46]
I'm having some issues with the plugin. It worked fine in my proof-of-concept tests, but when I started to build the "production" usage tests, it fails inconsistently.

The Airtable connection works, and the plugin successfully retrieves the new text (this is small sample with no proprietary information):

`Working on "Airtable tests ▸ Print page with notes ▸ Phone base layout ▸ Phone Layout ▸ Layout ▸ Main ▸ buttons ▸ 0 hero / default - text | hero-button-univ  ▸ 0 hero / default - icon_text | hero-button-univ  ▸ .*/univ/enclosed-button-univ ▸ Options ▸ .*/univ/content/text ▸ .*/univ/content/text_icon ▸ {{5-13-6.0}}"...
figma_app.592c5e6288ca461d485d7d3985c7c5bd.min.js:575   Replacing copy...
figma_app.592c5e6288ca461d485d7d3985c7c5bd.min.js:575   Original: 'PRIMARY BUTTON' , New: 'Placeholder Text'
figma_app.592c5e6288ca461d485d7d3985c7c5bd.min.js:575 Working on "Airtable tests ▸ Print page with notes ▸ Phone base layout ▸ Phone Layout ▸ Layout ▸ Main ▸ buttons ▸ 2 secondary / pressed - mb - text | secondary-button-univ ▸ 2 secondary / pressed - mb - icon_text | secondary-button-univ ▸ .*/univ/enclosed-button-univ ▸ Options ▸ .*/univ/content/text ▸ .*/univ/content/text_icon ▸ {{5-13-7.0}}"...
figma_app.592c5e6288ca461d485d7d3985c7c5bd.min.js:575   Replacing copy...
figma_app.592c5e6288ca461d485d7d3985c7c5bd.min.js:575   Original: 'Secondary Button' , New: 'Placeholder Text'`

However, the text in the Figma file does not actually change. (There seems to be some issue returning the text back to the node, maybe?) I get different errors (including 422) periodically, but this batch seems to be most useful:

`
      0x6716d0    1   0*       0x5541b8    Promise {  }.  jsvm-cpp.js.br:8
      0x7330a8    1   0*       0x518958      Error { message: 1"in set_characters: Expected \"characters\" to have type string but got array instead", stack: 1"    at f (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at next (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at Promise (native)\n    at s (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at forEach (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at next (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at Promise (native)\n    at h (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n" }.   jsvm-cpp.js.br:8
       0x670628    1   0*       0x5541b8    Promise {  }.  jsvm-cpp.js.br:8
       0x72f420    1   0*       0x518958      Error { message: 1"in set_characters: Expected \"characters\" to have type string but got array instead", stack: 1"    at f (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at next (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at Promise (native)\n    at s (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at forEach (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at next (native)\n    at <anonymous> (PLUGIN_3_SOURCE)\n    at Promise (native)\n    at h (PLUGIN_3_SOURCE)\n    at <anonymous> (PLUGIN_3_SOURCE)\n" }.  jsvm-cpp.js.br:8  
`

(The Airtable field is a long text type, and I've used long text type in the proof-of-concept file. I also tested everything I could think of on the Airtable side--table name format, field name formats, data types, key field construction, sorting, number of records, etc.)

I would appreciate any help you can give me. This plug-in is so close to solving a major process issue we're struggling with!

Thanks,
Dorothy

**Comment by imnotashrimp** [comment]
Hi - thanks for this! I'll have to have to look at it next week as we're heading right into a religious holiday for me. Would you be open to having a zoom call so I can better understand this?  @DorothyLindman

**Comment by DorothyLindman** [comment]
Sure, that would be great. I'll also keep poking at it this week to see if I can get any more information.

**Comment by DorothyLindman** [comment]
I think I figured out the issue!

In my proof- of-concept tests, I was syncing normal long-text fields.
In my "production" setup tests, I was syncing a LOOKUP field of a long-text field. Airtable only supports arrays in Lookup fields. I pasted the values from the column into a long text field, and it worked fine.

Now I'm looking into ways to convert the Lookup fields from arrays into text strings. But at least I have a workaround in the mean time.
